docs/ref/request-response.txt

```diff
@@ -554,11 +554,11 @@ a subclass of dictionary. Exceptions are outlined here:
 
 .. method:: QueryDict.__getitem__(key)
 
-    Returns the value for the given key. If the key has more than one value,
-    it returns the last value. Raises
-    ``django.utils.datastructures.MultiValueDictKeyError`` if the key does not
-    exist. (This is a subclass of Python's standard :exc:`KeyError`, so you can
-    stick to catching ``KeyError``.)
+    Returns the last data value for the given key. If the key has more than one
+    value, it returns the last value. If the value is an empty list, ``[]`` is
+    returned. Raises ``django.utils.datastructures.MultiValueDictKeyError`` if
+    the key does not exist. (This is a subclass of Python's standard
+    :exc:`KeyError`, so you can stick to catching ``KeyError``.)
```

Comment:
```suggestion
    Returns the last value for the given key; or an empty list (``[]``) if the
    key exists but has no values. Raises
    ``django.utils.datastructures.MultiValueDictKeyError`` if the key does not
    exist. (This is a subclass of Python's standard :exc:`KeyError`, so you can
    stick to catching ``KeyError``.)

    .. code-block:: pycon

        >>> q = QueryDict("a=1&a=2&a=3", mutable=True)
        >>> q.__getitem__("a")
        '3'
        >>> q.__setitem__("b", [])
        >>> q.__getitem__("b", [])
        []
```
I agree with @shangxiao that it's currently repetitive. I would also consider an example to be explicit.

Reply:
Hi, I see the example adheres to the convention in the docs. https://docs.djangoproject.com/en/5.1/ref/request-response/

An example to be explicit is a good idea,

Reply:
Yes, if you agree with a suggestion commit it and squash all commits to have a single commit and update the commit message to match the format e.g. `Fixed #35915 -- Clarified the empty list case in QueryDict.__getitem__() docs.`
Sometimes mergers do some commit cleanup (I am a merger :wave:) and so in this case, I have pushed up the tweak :+1: